test/models.ts

```diff
@@ -0,0 +1,39 @@
+import type { Random } from "@reside-ic/random";
+
+import type { base, BaseType } from "../src/base";
+import type { DustModel, DustModelInfo } from "../src/model";
+import type { InternalStorage, ParsType } from "../src/pars";
+
+// In contrast with the equivalent models in odin-js I'm leaving these
+// as TypeScript (not JS) for now to help get types right. That does
+// mean that there's a bit of a fight with extracting things from the
+// InternalStorage type.
```

Comment:
Might be worth having more specific InternalStorage type for models which will only ever use simple numerics like this one? Particularly if it's just being used internally and isn't required for the interface. Oh, I guess you need it to be InternalStorage to pass to setParScalar? But might be less annoying to do a cast there rather than everywhere else?

Reply:
This approach mirrors exactly the odin-js approach in order to make the generation easy - and normally there will be no types there as we just emit js and not ts. However, I've added types here in the test models to help me refactor as I went